src/sentry/static/sentry/app/views/releaseDetails.jsx

```diff
@@ -18,6 +21,53 @@ import TimeSince from 'app/components/timeSince';
 import Version from 'app/components/version';
 import withEnvironmentInQueryString from 'app/utils/withEnvironmentInQueryString';
 
+class ReleaseDetailsActions extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+    orgId: PropTypes.string.isRequired,
+    projectId: PropTypes.string.isRequired,
+    release: PropTypes.object.isRequired,
+  };
+
+  onDelete() {
+    let {orgId, projectId, release, api} = this.props;
+    let path = `/organizations/${orgId}/releases/${release.version}/`;
+    let redirectPath = `/${orgId}/${projectId}/releases/`;
+    let indicator = IndicatorStore.add(t('Deleting Release...'));
+
+    api.request(path, {
+      method: 'DELETE',
+      success: () => {
+        IndicatorStore.remove(indicator);
+        browserHistory.push(redirectPath);
+      },
+      error: err => {
+        IndicatorStore.remove(indicator);
+        IndicatorStore.addError(t(err.responseJSON.detail));
+      },
+    });
+  }
+
+  render() {
+    return (
+      <div className="release-actions m-b-1">
+        <div className="btn-group">
+          <LinkWithConfirmation
+            className="release-remove btn btn-default btn-sm"
+            title={t('Delete')}
+            message={t(
+              'Deleting this release is permanent. Are you sure you wish to continue?'
+            )}
+            onConfirm={this.onDelete.bind(this)}
+          >
+            <span className="icon-trash" style={{marginRight: 5}} />Delete
```

Comment:
I'm not a fan of the inline styles here. This approach was used on the Issues overview actions as well. I plan on revisiting this soon.